### ISSUE TITLE
Error updating the git index

### ISSUE THREAD

**Issue by akraines** [git-lfs/git-lfs#2937]
I'm having alot of trouble pulling my git lfs enabled repo from bitbucket on my Teamcity docker build agent. I setup git-lfs correctly AFAIK.
My machine is working with git-lfs correctly. also, before I initialised git-lfs on the server, it pulled all the git files correctly - just that the build was missing all of its media artifacts :-). So I installed git-lfs and reran the build and about 50% of the way downloading the files it failed the pull.
I logged into the server and ran  'git lfs pull' manually and it gave me the follwing error:
`Error updating the git index: (0/1), 0 B | 0 B/s
error: media/toolbox.png: cannot add to the index - missing --add option?
fatal: Unable to process path media/toolbox.png
`

`
LFS: Client error: https://api.media.atlassian.com/file/e347315c-f605-4cc2-813c-c44a8b6a3d4a/binary
error: failed to fetch some objects from 'https://bitbucket.org/theguggenheimgroup/tanach-machine.git/info/lfs'
`

**Comment by ttaylorr** [comment]
Hi @akraines, thanks for opening this issue and sorry that you're having trouble! I haven't seen this error before; can you run the same command that caused it with `GIT_TRACE=1` in your environment? If you remove any sensitive credentials, posting the output here would be helpful in pinning this down. Thanks!

**Comment by lukasholzer** [comment]
Hey, I've got a similar problem.
I want to pull a file from a git repository, created with `--no-checkout` flag

```bash
# 1) Step create dir
mkdir $foldername
cd $foldername

# 2) checkout
git clone --no-checkout --depth=100 -b $BRANCH_NAME $GIT_URL ./workdir

# 3) fetch origin
cd workdir
git fetch origin master

# 4) git pull only one file (this stage fails) 
git lfs pull --include=all_components_master_file/01_all_components_library.sketch
```

This last command returns the following error.
The weird thing is, that the pull of the file works and the file is in my storage if I check it with `ls -lah`
but it fails with a non-zero exit code.
@ttaylorr If I execute it with GIT_TRACE=1 nothing happens.

```bash
Error updating the git index:
error: all_components_master_file/01_all_components_library.sketch: cannot add to the index - missing --add option?
fatal: Unable to process path all_components_master_file/01_all_components_library.sketch


Errors logged to /Users/lukas.holzer/Sites/test/git-lfs-pull-fix/workdir/.git/lfs/logs/20180717T093056.589977176.log
Use `git lfs logs last` to view the log.

```

**Comment by akraines** [comment]
My issue was a ssl certificate issue, which is now solved. I'll leave this issue open for lukasholzer to close when he is done.

**Comment by ttaylorr** [comment]
I haven't seen this one before, either. @larsxschneider do you have any ideas?

**Comment by larsxschneider** [comment]
@lukasholzer I think the problem is the `--no-checkout` option. Since you do not checkout, there is no index file (== Git's internal representation of the current checkout). Consequently, Git LFS cannot operate on this index file.

We might be able to teach Git LFS to work around this ... however what is your greater goal with these commands?

**Comment by ttaylorr** [comment]
> > @lukasholzer I think the problem is the `--no-checkout` option.
> > Since you do not checkout, there is no index file (== Git's internal
> > representation of the current checkout). Consequently, Git LFS
> > cannot operate on this index file.
>
> We might be able to teach Git LFS to work around this ... however what
> is your greater goal with these commands?

Thanks, I'm interested in this as well.

**Comment by shadeslayer** [comment]
Hi!
I'm also interested in this feature. The greater goal for me is to be able to store OpenGL traces and checkout individual traces in the git repository as opposed to checking them all out in the initial clone. This would be very useful when storage is constrained resource and we only need one file at a time.

**Comment by WangYuandong** [comment]
Hi, I have also got the same "Error updating the git index" problem. 

The error is the following.
###
Error updating the git index: (33945/33945), 4.1 GB | 14 MB/s
error: android-ndk-r10d/platforms/android-21/arch-x86/usr/include/OMXAL/OpenMAXAL.h: cannot add to the index - missing --add option?
fatal: Unable to process path android-ndk-r10d/platforms/android-21/arch-x86/usr/include/OMXAL/OpenMAXAL.h

Errors logged to test/.git/lfs/logs/20191116T003405.394447119.log
Use `git lfs logs last` to view the log.
###

I just want to pull a git repository with git LFS enabled.

### 
#1) git clone $GIT_URL

#2) git config remote.origin.lfsurl <artifactory_repo_url>
git config lfs.<lfsurl>.locksverify false

#3) git lfs pull (this stage failed with this problem)

###

I'm using git-2.19.1 and git-lfs-2.5.2 on ubuntu 18.10.
Is there any idea about how to find the root cause?

**Comment by bk2204** [comment]
Hey,

Are you also using a bare repository or one without a checkout in some way?  Does `.git/index` exist?  Do you have any files with a newline in their names?  Does this work with a newer version of Git LFS?

**Comment by WangYuandong** [comment]
Hey, @bk2204 

There are no any files with a newline in their names. All file names are regular. 
When this problem appeared, there are no `.git/index` file existing.

If I use "git clone" without a checkout and checkout it manually after configuring LFS, this error is gone.

1 `git clone --no-checkout $git_repo_url`

2 Configure git LFS manually as previous 

3 `git fetch origin master`

4 `git checkout master`

There are no previous error and `.git/index` file also exist.

So it seems to be that if git repo is checkout without LFS configured, the `.git/index` file cannot be created as normal and so `git lfs pull` returns "updating git index error". 

Is this conclusion right?

**Comment by bk2204** [comment]
I don't think that it's that the repo is checked out without LFS configured but that the clone happened with `--no-checkout`, which means that there's no index created.  If you want to use `git lfs checkout` or `git lfs pull` (which is equivalent to `git lfs fetch` and `git lfs checkout`), you need to have a checkout in the first place, since checking out files necessarily requires an index.

We don't create the index: it's Git that does this, so we need it to have done that by the time we check things out.

What you could try to do is run your clone with `GIT_LFS_SKIP_SMUDGE=1` and then configure your custom LFS settings and run `git lfs checkout`, which might avoid some of the pain.

**Comment by WangYuandong** [comment]
@bk2204 Got it. `GIT_LFS_SKIP_SMUDGE=1` can help avoid the error indeed.

**Comment by mrcaron** [comment]
I'm seeing this when trying to do an LFS pull of one file:

`git lfs pull -I path/to/file/giving/lfs/error`

Then I see:

    Error updating the git index: (1/1), 2.5 KB | 0 B/s
    error: path/to/file/giving/lfs/error: cannot add to the index - missing --add option?
    fatal: Unable to process path sldworks/AMRES/swstyles/swstyleblue/res2/toolbars/routing/routingtools/i_routingroutingtools_40.png

There's no `--add` option for LFS pull. I'm not sure how to resolve this issue. And I haven't used the --no-checkout when cloning.

**Comment by bk2204** [comment]
@mrcaron This sounds like a separate issue.  Could you open a new issue, fill out the issue template, and include reproduction steps?  That would help us figure out where the problem is.

**Comment by Timon0305** [comment]
Please check if there is another git file.

**Comment by RosenGeorgievvv** [comment]
Hello @bk2204

I have faced similar issue, which i want to report.
I've created a repository through GitHub Desktop version. I have successfully connected my VSC with GitHub repo.
However, I noticed something interesting:

![image](https://user-images.githubusercontent.com/73111903/143466897-c108eac8-fc14-4e15-ac0f-9f6aaa3d19aa.png)

In the above screenshot of my screen, you could see when i try to commit the changes, I receive the following error message.
From the VSC you could see my client folder, in GitHub Desktop it appears like: cclient ( which it shouldn't ). 
I have tried to search the reported error message, but without any luck

**Comment by chrisd8088** [comment]
Hi @RosenGeorgievvv -- if you have a problem with GitHub Desktop and/or Visual Studio Code, I would report those to the relevant support teams, as this is just a repo for the Git LFS client.

If you have a specific problem with the Git LFS client, though, please do open a fresh issue and fill out all the details in the template (reproduction steps, `git lfs env` output, etc.)  Thanks and good luck!

**Comment by amitgoyal1310** [comment]
> 

Did you manage to resolve this issue. I am stucked with 22 GB repo with lfs. and the same error that you are getting. I am using. 
git-lfs/2.9.0 (GitHub; windows amd64; go 1.12.7; git 8ab05aa7)
git version 2.10.2.windows.1

**Comment by Bra1nsen** [comment]
![image](https://github.com/git-lfs/git-lfs/assets/58213050/3a188190-0c0b-419a-86e3-e50ee5e30fba)

No there is not second .git folder, what now??
